Improve mouse keys docs and constant speed mode

Markus00000 opened the pull request:
## General changes

* Make default speed settings more consistent across modes:
    * In accelerated and constant+momentary mode, one setting is faster than unmodified speed and two settings are slower
    * In constant+tap-to-select mode, there is no unmodified speed
* Improve documentation

## Changes to constant speed modes

* Add an unmodified speed that is used in momentary mode when no `KC_ACLx` key is pressed
    * This gives users another speed setting without introducing new key codes
    * Previously, one key code (`KC_ACL1`) had no effect because its speed was identical to the unmodified speed, which could confuse users and made one key code useless
* The speeds are named `0`, `1`, `2`, thus corresponding to their respective key codes `KC_ACL0`, `KC_ACL1`, `KC_ACL2`
    * Easier to configure as one no longer needs to translate between speed descriptions (“SLOW”) and numbers (`KC_ACL0`) and, for example, a user’s fastest speed can no longer have `SLOW` in its variable name
    * Furthermore, the `SLOW`, `MEDIUM`, `FAST` naming scheme did not indicate where the unmodified speed in momentary mode should be placed in the series
* Use sensible default settings (YMMV…)

(The original PR has since been merged. Consequently, the following description is no longer useful.)

~~This recreates pull request #2246 from @JonArintok (unresolved conflicts) with few changes.~~

~~Mousekeys can be configured to use three different, constant speeds for cursor movements and wheel scrolling.~~

~~(I am not sure if the long if-else blocks in `mousekey.c` and `mousekey.h` should be refactored.)~~

skullydazed commented on docs/feature_mouse_keys.md:
```diff
@@ -21,30 +22,36 @@ You can see an example here: https://github.com/qmk/qmk_firmware/blob/master/key
 ## Configuring the Behavior of Mousekeys
 
-The default speed for controlling the mouse with the keyboard is intentionally slow. You can adjust these parameters by adding these settings to your keymap's `config.h` file. All times are specified in milliseconds (ms).
+Regardless of whether you use cursor acceleration or constant speeds, all times are specified in milliseconds (ms).
```
Could you add some introductory text before this? Something like, "There are two movement models you can select between for mousekeys. Cursor acceleration [short description]. By contrast, Constant Cursor [short description]."

skullydazed commented on docs/feature_mouse_keys.md:
```diff
@@ -55,27 +62,55 @@ The default speed for controlling the mouse with the keyboard is intentionally s
+
+You can define three different speeds (slow, medium, fast) for both the cursor and the mouse wheel, and also whether speed selection is momentary or tap-to-select.
+
+For each speed you can specify the interval between reports in milliseconds and how many pixels you want the cursor to move per report (offset).
```
Small nitpick: could this be part of the previous sentence to form a paragraph?

skullydazed commented on docs/feature_mouse_keys.md:
````diff
@@ -55,27 +62,55 @@ The default speed for controlling the mouse with the keyboard is intentionally s
 #define MOUSEKEY_WHEEL_TIME_TO_MAX 40
 ```
````
I know this was already here, but since we're touching this documentation could you reformat this as a table with the following headers?

    Define, Default, Description

You can fold the descriptions below into it and make everything more compact and easier to read/find.

skullydazed commented on docs/feature_mouse_keys.md:
```diff
@@ -55,27 +62,55 @@ The default speed for controlling the mouse with the keyboard is intentionally s
+#define MK_W_OFFSET_MED      1
+#define MK_W_INTERVAL_MED  200
+#define MK_W_OFFSET_FAST     1
+#define MK_W_INTERVAL_FAST 100
```
As above, could this be reformatted into a table to follow the format the rest of the docs are moving to?

Markus00000 commented:
@skullydazed I have incorporated your suggestions.

I have made numerous changes to the documentation and renamed `MK_MOMENTARY_ACCEL` to `MK_MOMENTARY_SPEED` as there is no acceleration involved.

Please let me know if you notice any issues or if you have suggestions.

Markus00000 commented:
## General changes

* Use consistent default speed settings for both modes:
    * One setting is faster than standard speed
    * Two settings are slower than standard speed
* Improved documentation of both modes

## Changes to constant speed mode

* Add a standard speed that is used when no `KC_ACLx` key is pressed
    * This gives users another speed setting without introducing new key codes
    * Previously, one key code (`KC_ACL1`) had no effect because its speed was identical to the unmodified speed, which could confuse users and made one key code useless
* The defaults for both cursor movement speeds and scroll speeds are:
    * `KC_ACL0` doubles standard speed
    * `KC_ACL1` reduces standard speed by one third
    * `KC_ACL2` reduces standard speed by another third
* The speeds are named `0`, `1`, `2`, thus corresponding to their respective key codes `KC_ACL0`, `KC_ACL1`, `KC_ACL2`
    * Easier to configure as one no longer needs to translate between speed descriptions (“SLOW”) and numbers (`KC_ACL0`) and, for example, a user’s fastest speed can no longer have `SLOW` in its variable name
    * Furthermore, the `SLOW`, `MEDIUM`, `FAST` naming scheme did not indicate where the default speed should be placed in the series
